packages/salesforcedx-vscode-apex/src/index.ts

```diff
@@ -315,6 +312,26 @@ async function createLanguageClient(extensionContext: vscode.ExtensionContext) {
 
     languageClientUtils.setClientInstance(languageClient);
     await languageClient!.start();
+    // Client is running
+
+    // Listener is useful only in async mode
+    if (!languageServer.enableSyncInitJobs) {
+      // The listener should be set after languageClient is ready
+      // Language client will get notified once async init jobs are done
+      languageClient.onNotification(API.doneIndexing, async () => {
+        await getTestOutlineProvider().refresh();
+        languageServerReady();
+      });
+    } else {
+      // Client will check status with server after initialization (only for sync mode)
+      // In async mode, indexerStatus is never RUNNING at the point
+      const indexerStatus = await languageClient.sendRequest(API.indexerStatus);
```

Comment:
@mingxuanzhangsfdx Given
* we reach here only in sync mode
* which means indexer should have done its work (running)

Why do we need to verify that it is in running?